manager/apis/v2/handler/error.go

```diff
@@ -1,18 +1,52 @@
 package handler
 
-import "github.com/gin-gonic/gin"
+import (
+	"fmt"
+	"net/http"
+
+	"d7y.io/dragonfly/v2/pkg/dferrors"
+	"github.com/gin-gonic/gin"
+)
+
+func (handler *Handler) Error(ctx *gin.Context) {
+	ctx.Next()
+	e := ctx.Errors.Last()
+	if e != nil {
+		err, ok := e.Err.(*HTTPError)
+		if !ok {
+			ctx.JSON(http.StatusNotFound, err)
+			return
+		}
+		ctx.JSON(err.Code, err)
+	}
+}
 
 // NewError example
-func NewError(ctx *gin.Context, status int, err error) {
-	er := HTTPError{
-		Code:    status,
+func NewError(status int, err error) error {
+	er := &HTTPError{
 		Message: err.Error(),
 	}
-	ctx.JSON(status, er)
+
+	if status > 0 {
+		er.Code = status
+		return er
+	}
+	e, ok := err.(*dferrors.DfError)
```

Comment:
i dont't kown the benefit of the refactor. 
er.Code = int(e.Code) ,   it's not suitable to set internal error code as http error code

Reply:
@yangjun289519474 yes, you are right. I  will fix it. I am so sorry. I didn't pay attention to this.